kuba-kostrzewski/simpleToDoApp/src/main/java/pl/kostrzej/simpleToDoApp/service/UserService.java

```diff
@@ -0,0 +1,75 @@
+package pl.kostrzej.simpleToDoApp.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.kostrzej.simpleToDoApp.model.User;
+import pl.kostrzej.simpleToDoApp.repository.UserRepository;
+
+import java.util.Scanner;
+
+@Service
+public class UserService {
+
+    UserRepository userRepository;
+    Scanner scanner;
+
+    @Autowired
+    public UserService(UserRepository userRepository, Scanner scanner) {
+        this.userRepository = userRepository;
+        this.scanner = scanner;
+    }
+
+    public void addUser(){
+        User user = new User();
+        String login, password, question, answer;
+        do {
+            System.out.println("Podaj nazwę użytkownika:");
+            login = scanner.nextLine();
+        } while (checkLogin(login));
+        user.setLogin(login);
+        do {
+            System.out.println("Podaj hasło:");
+            password = scanner.nextLine();
+        } while (checkPassword(password));
+        user.setPassword(password);
+        System.out.println("Podaj e-mail lub zostaw puste:");
+        user.setEmail(scanner.nextLine());
+        do {
+            System.out.println("Podaj pytanie pomocnicze:");
+            question = scanner.nextLine();
+        } while (isFieldEmpty(question, "Pytanie pomocnicze"));
+        user.setQuestion(question);
+        do {
+            System.out.println("Podaj odpowiedź:");
+            answer = scanner.nextLine();
+        } while (isFieldEmpty(answer, "Odpowiedź"));
+        user.setAnswer(answer);
+        userRepository.save(user);
+        System.out.println("Użytkownik został poprawnie zarejestrowany\n" +
+                "Możesz się teraz zalogować");
+    }
+    private boolean checkPassword(String password){
```

Comment:
Metody z przedrostkiem check nie są dla mnie zrozumiałe na pierwszy rzut oka - muszę się wczytywać kiedy zwracają true, a kiedy false. Myślałem, że zwracają true kiedy jest ok, lecz u Ciebie jest inaczej. Doszedłem do tego dopiero po przeczytaniu kodu, a tak być nie powinno.

Zamień metody np. na isLoginCorrect(String login) lub w drugą strone: incorrect. Będzie to czytelniejsze tak jak isFieldEmpty.

Reply:
poprawione zostały wszystkie metody sprawdzające poprawność danych